0015-helm-value-style-variable-interfaces/README.md

```diff
@@ -0,0 +1,208 @@
+
+**Given** I have a Zarf Package with a Helm value override in a chart
+**When** I deploy that package with a `zarf-config.yaml` like the below*:
+```yaml
+package:
+  deploy:
+    set:
+      MY_OVERRIDE_VARIABLE_NAME:
+        key: value
+        bool: true
+        num: 100
+        arr: []
+        obj: {}
+```
+**Then** Zarf will apply the entire interface to the Helm chart override
+
+> [!NOTE]
+> *This would apply to all `zarf-config` formats not just YAML
+
+#### Story 2
+
+**As** Jacquline **I want** to be able to set full Helm value objects from the CLI **so that** I can simplify package configurations and rely on my existing familiarity with Helm.
+
+**Given** I have a Zarf Package with a Helm value override in a chart
+**When** I deploy that package with a `--set` like the below:
+```yaml
+zarf package deploy zarf-package-test.tar.zst --set MY_OVERRIDE_VARIABLE_NAME.key=value,MY_OVERRIDE_VARIABLE_NAME.bool=true,MY_OVERRIDE_VARIABLE_NAME.num=100,MY_OVERRIDE_VARIABLE_NAME.arr=[]
+```
+**Then** Zarf will apply the entire interface to the Helm chart override
+
+> [!NOTE]
+> *This should follow the [Helm `--set` syntax](https://helm.sh/docs/intro/using_helm/#the-format-and-limitations-of---set)
+
+### Risks and Mitigations
+
+This will require some additional processing of the `zarf-config` files to allow them to be processed properly because of Viper's long-running [case-insensitive keys issue](https://github.com/spf13/viper/issues/1014).  This could be performed similar to UDS CLI's implementation of this feature and / or by deprecating some of the existing less-used formats for configuration.
+
+The `--set` syntax will change somewhat how variables are interpreted on the CLI (i.e. `--set VAR=100` will no longer represent `"100"` and instead will just be `100` internally). For `###` templates this can be mitigated by simply representing the value as a string for backwards compatibility though this will likely need to either be marked as breaking for Helm overrides or have additional flag changes / opt ins for this specific feature.
+
+## Design Details
+
+This design proposal keeps changes to a minimum to align Zarf Variables with Helm Values with the largest changes being the change from `string` to `interface` and the changes to loading a `zarf-config` file and handling `--set` as described above.
+
+### Test Plan
+
+[X] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this proposal.
+
+##### Prerequisite testing updates
+
+NA - This is a modification of existing behavior that should not require prerequisite testing updates.
+
+##### Unit tests
+
+Variable interfaces and libraries should be updated to ensure that interfaces are properly handled as opposed to strings.
```

Comment:
I cannot imagine this functionality being added without extensive fuzzy tests, ensuring that we're not able to break zarf injecting various random strings, including to potential code injections.

Reply:
added this in the pre-req section since it would be a new type of testing (I believe) and also added some links to how Helm handles this internally